src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,117 @@
+import duke.tasks.Todo;
+
+
+/**
+ * Parses in the user input and executes the corresponding actions
+ */
+public class Parser {
+    private final String command;
+    private final Ui ui;
+
+    /**
+     * Constructor for parser
+     * @param command user's inputted command
+     */
+    public Parser(String command) {
+        this.command = command;
+        this.ui = new Ui();
+    }
+
+    /**
+     * Parses the command and matches it with the corresponding action to be taken
+     * Executes the action on the task list given
+     * @param tasks ArrayList of current tasks stored on the chatbot
+     * @return boolean true if bye command entered and chatbot should be stopped, false otherwise
+     */
+    public String executeCommand(TaskList tasks, Storage storage) throws IOException {
+        String[] commands = command.strip().split(" ", 2);
+        try {
+            switch (commands[0]) {
+            case "bye":
+                storage.saveTasks(tasks);
+                return ui.showBye();
+            case "list":
+                return ui.getCorrectMessage(Ui.Commands.LIST, tasks, 0);
+            case "mark":
+                if (commands.length == 1) {
+                    throw new EmptyDescriptionException("mark");
+                }
+                int indexMark = Integer.parseInt(commands[1]) - 1;
+                tasks.markIndex(indexMark);
+                return ui.getCorrectMessage(Ui.Commands.MARK, tasks, indexMark);
+            case "unmark":
+                if (commands.length == 1) {
+                    throw new EmptyDescriptionException("unmark");
+                }
+                int indexUnmark = Integer.parseInt(commands[1]) - 1;
+                tasks.unmarkIndex(indexUnmark);
+                return ui.getCorrectMessage(Ui.Commands.UNMARK, tasks, indexUnmark);
+            case "todo":
+                if (commands.length == 1) {
+                    throw new EmptyDescriptionException("todo");
+                }
+                tasks.add(new Todo(commands[1]));
+                return ui.getCorrectMessage(Ui.Commands.TASK, tasks, tasks.getSize() - 1);
+            case "event":
+                if (commands.length == 1) {
+                    throw new EmptyDescriptionException("event");
+                }
+                String[] eventCommands = commands[1].split(" /at ", 2);
+                if (eventCommands.length == 1) {
+                    return ui.getErrorMessage("datetime");
+                }
+                try {
+                    LocalDate timeStringParsed = LocalDate.parse(eventCommands[1]);
+                    tasks.add(new Event(eventCommands[0],
+                            timeStringParsed.format(DateTimeFormatter.ofPattern("MMM dd yyyy"))));
+                    return ui.getCorrectMessage(Ui.Commands.TASK, tasks, tasks.getSize() - 1);
+                } catch (DateTimeParseException error) {
+                    return ui.getErrorMessage("datetime");
```

Comment:
Perhaps you can also write a separate method to check for the validity of a date provided by the user, as stated in this [section](https://nus-cs2103-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html#minimize-code-duplication) of the textbook